tests/typecheck/managers/querysets/test_annotate.yml

```diff
@@ -395,3 +395,22 @@
 
         class Entry(models.Model):
             blog = models.ForeignKey(Blog, on_delete=models.CASCADE)
+
+- case: test_annotate_allows_any_lookups_in_filter
+  main: |
+    from django.db import models
+    from myapp.models import Blog
+
+    qs = Blog.objects.annotate(distance=0)
+    qs.filter(distance=10)
+    qs.filter(distance__lt=10)
+    qs.filter(distance__lt__lt=10)
+    qs.filter(distance__unknown_lookup=10)
```

Comment:
Could we reveal `qs` and 1 `filter` at least so we know we're working with an expected type here (and not `Any` or something swallowing whatever)